continuous_integration/environment-mindeps-optional.yaml

```diff
@@ -0,0 +1,54 @@
+name: test-environment
+channels:
+  - conda-forge
+dependencies:
+  # required dependencies
+  - packaging=20.0
+  - python=3.8
+  - pyyaml=5.3.1
+  - click=7.0
+  - cloudpickle=1.1.1
+  - partd=1.2.0
+  - fsspec=2021.09.0  # required for arrow I/O support
+  - importlib-metadata=4.13.0
+  - toolz=0.10.0  # required for min version of cytoolz
+  # optional dependencies pulled in by pip install dask[array,dataframe]
+  - numpy=1.21
+  - pandas=1.3
+  # optional dependencies used by dask
+  - bokeh=2.4.2
+  - cachey=0.1.1
+  - crick=0.0.3
+  - cytoolz=0.10.1
+  - fastavro=0.22.6
+  - fastparquet=0.7.0
+  - h5py=2.10.0
+  - ipycytoscape=1.0.1
+  - IPython=7.9.0
+  - jinja2=2.10.3
+  - lz4=3.1.1
+  - matplotlib=3.1.1
+  - mimesis=4.0.0
+  - psutil=5.6.3
+  - pyarrow=7.0
+  - python-graphviz=0.8.4
+  - python-snappy=0.5.4  # required for default compression algorithm for to_parquet
+  - s3fs=0.1.3
```

Comment:
In other CI builds we've been using this restriction

https://github.com/dask/dask/blob/2c8296c80ec436cc8a7fef98a2832a9032b287ff/continuous_integration/environment-3.9.yaml#L53-L55

`2021.8.0` has been out for a while. Let's just bump the minimum to `2021.8.0` and remove those version restrictions in the other CI environment files.

Reply:
It looks like `s3fs=2021.8.0` is incompatible with `fsspec=2021.9.0` due to a tightly pinned dependency:

```
Could not solve for environment specs
The following packages are incompatible
├─ fsspec 2021.09.0**  is requested and can be installed;
└─ s3fs 2021.8.0**  is uninstallable because it requires
   └─ fsspec 2021.7.0 , which conflicts with any installable versions previously reported.
```

Any opposition to bumping the minimum to `2021.9.0` to circumvent this?

Reply:
> Any opposition to bumping the minimum to 2021.9.0 instead to circumvent this?

Not at all, sounds good 👍